.github/workflows/build.yml

```diff
@@ -185,7 +185,7 @@ jobs:
   tests:
     needs: prepare
     runs-on: ubuntu-20.04
-    timeout-minutes: 25
+    timeout-minutes: 45
```

Comment:
this seems excessive. if there are tests that take >25 minutes to run we should rewrite them rather than this.

i'm working on improvements for the prep/test times. as this doesn't seem to help with the overall issue (tests never finish due to some unknown reason) I say remove this for now.

Reply:
I was seeing random failures of various tasks. In two cases, tests were timing out, not the same tests each time. In one case formatting was timing out. And then there was some 40 min timeout, I could explain.

So, I don't think the tests as such were to blame.

But I agree 45 min is a bit excessive, but I was just tired of rerunning the build without success.

Reply:
And I really need to get this build through the pipeline.

Reply:
I get the frustration. But it looks to me to have been extremely random - at least none of the test steps in your latest run exceeded 16 minutes.

The good news is that looking at this reinforced me in my belief that we need to re-think how we're doing integration tests, and that's next on my to-do list. The bad news is it might take some days.

Bottom line - do what you need to do to get this through for now, and the future will be brighter... :sun_behind_large_cloud: